WireAPI/Sources/WireAPI/APIs/UsersAPI/UsersAPIV0.swift

```diff
@@ -0,0 +1,149 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class UsersAPIV0: UsersAPI {
+
+    let httpClient: HTTPClient
+    let decoder = ResponsePayloadDecoder(decoder: .defaultDecoder)
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    var basePath: String {
+        switch apiVersion {
+        case .v0:
+            ""
+        default:
+            "/v\(apiVersion.rawValue)"
+        }
+    }
```

Comment:
A late change to the `TeamsAPI` pr removed the `apiVersion` property and instead had an explicit override of the base path in every version, like:

```swift
var basePath: String {
    "/users"
}

override var basePath: String {
    "/v1/users"
}

override var basePath: String {
    "/v2/users"
}
```

The reasoning behind this was to be more explicit with the path. Perhaps we should also do that here? What do you think?

Reply:
The path isn't always `/users` it's also `/list-users`. What's the benefit about being explicit about the path? I would be duplicating this for all api versions.

Reply:
Well I don't understand why I should care about the version prefix for every api we implement, it always follows the same pattern.

Maybe we should have some base API class which provides the `versionPathPrefix`

Reply:
I introduced a `VersionedAPI` protocol to reduce the code depublication around this. I you agree I would also like to delete the `basePath(teamID:)` function from the TeamAPI classes.

Reply:
Let's have a quick group chat about it